### PR TITLE
Moves to supervisor rather than application for TaskBunny startup

### PR DESCRIPTION
**Pull request by britton-jb**

### CODE REVIEW

**Review by kalolo — APPROVED** [review]
May need more context to understand if `register_metrics` still getting call by the start_link .. but I guess it is .. so aside from that changes makes sense to me.

File: lib/mix/tasks/task_bunny.queue.reset.ex [hunk 1]
```diff
@@ -1,10 +1,10 @@
 defmodule Mix.Tasks.TaskBunny.Queue.Reset do
```
Comment (britton-jb) [inline lib/mix/tasks/task_bunny.queue.reset.ex:1]: This needs reworked still.

File: lib/task_bunny/supervisor.ex [hunk 2]
```diff
@@ -5,59 +5,47 @@ defmodule TaskBunny.Supervisor do
...
   @doc false
-  @spec start_link(atom, atom) :: {:ok, pid} | {:error, term}
-  def start_link(name \\ __MODULE__, wsv_name \\ WorkerSupervisor, ps_name \\ :publisher) do
-    Supervisor.start_link(__MODULE__, [wsv_name, ps_name], name: name)
+  @spec start_link(Keyword.t()) :: {:ok, pid} | {:error, term}
+  def start_link(opts) do
+    name = Keyword.get(opts, :name, TaskBunny.Supervisor)
+    workers = Keyword.get(opts, :workers, [])
+    publisher = Keyword.get(opts, :publisher, {:publisher, []})
+    Supervisor.start_link(__MODULE__, [workers, publisher], name: name)
```
Comment (britton-jb) [inline lib/task_bunny/supervisor.ex:19]: Not quite through the renaming process throughout the codebase.

File: lib/task_bunny/worker_supervisor.ex [hunk 3]
```diff
@@ -10,26 +10,45 @@ defmodule TaskBunny.WorkerSupervisor do
...
   @doc false
-  @spec start_link(atom) :: {:ok, pid} | {:error, term}
-  def start_link(name \\ __MODULE__) do
-    Supervisor.start_link(__MODULE__, [], name: name)
+  @spec start_link(Keyword.t()) :: {:ok, pid} | {:error, term}
+  def start_link(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    queues = Keyword.get(opts, :queues, [])
+    # FIXME do we want this here?
+    Enum.each(queues, &Initializer.declare_queue/1)
```
Comment (britton-jb) [inline lib/task_bunny/worker_supervisor.ex:21]: Open question.

File: test/task_bunny/supervisor_test.exs [hunk 4]
```diff
@@ -55,9 +62,11 @@ defmodule TaskBunny.SupervisorTest do
     :ok
   end
 
+  #  FIXME have somehow broke this test
```
Comment (britton-jb) [inline test/task_bunny/supervisor_test.exs:65]: Need to take another look at this one.

File: test/task_bunny/worker_supervisor_test.exs [hunk 5]
```diff
@@ -1,133 +1,143 @@
...
-      WorkerSupervisor.graceful_halt(pid, 100)
-
-      :timer.sleep(1_100)
-
-      %{message_count: count} =
-        Queue.state(
-          Connection.get_connection!(),
-          @queue
-        )
+# defmodule TaskBunny.WorkerSupervisorTest do
```
Comment (britton-jb) [inline test/task_bunny/worker_supervisor_test.exs:1]: Need to address this file still.

**Pull request closed without merging**